src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/PSWorker.java

```diff
@@ -50,66 +48,68 @@
 	private MatrixObject _valFeatures;
 	private MatrixObject _valLabels;
 
-	ArrayList<DataIdentifier> _outputs;
+	DataIdentifier _output;
 	FunctionCallCPInstruction _inst;
 
 	public PSWorker(long workerID, String updFunc, Statement.PSFrequency freq, int epochs, long batchSize,
-			ListObject hyperParams, ExecutionContext ec, ParamServer ps) {
+			ExecutionContext ec, ParamServer ps) {
 		this._workerID = workerID;
 		this._updFunc = updFunc;
 		this._freq = freq;
 		this._epochs = epochs;
 		this._batchSize = batchSize;
-		this._ec = ExecutionContextFactory.createContext(ec.getProgram());
-		if (hyperParams != null) {
-			this._ec.setVariable(Statement.PS_HYPER_PARAMS, hyperParams);
-		}
+		this._ec = ec;
 		this._ps = ps;
```

Comment:
Avoid unnecessary `this.` - the naming schemes of a "_"-prefix for instance members makes this redundant.